library/std/arrays.qs

```diff
@@ -985,6 +1033,54 @@ namespace Microsoft.Quantum.Arrays {
         array[... -1 ...]
     }
 
+    /// # Summary
+    /// Rotates an array left or right by a specific step size.
+    ///
+    /// # Type Parameters
+    /// ## 'T
+    /// The type of the array elements.
+    ///
+    /// # Input
+    /// ## stepCount
+    /// The amount of positions by which the array elements will be shifted.
+    /// If this is positive, `array` is rotated to the right.
+    /// If this is negative, `array` is rotated to the left.
+    /// ## array
+    /// Array to be padded.
+    ///
+    /// # Output
+    /// An array `output` that is the `array` rotated to the right or left
+    /// by the specified step size.
+    ///
+    /// # Example
+    /// ```qsharp
+    /// let array = [10, 11, 12];
+    /// // The following line returns [11, 12, 10].
+    /// let output = Rotated(2, array);
+    /// // The following line returns [12, 10, 11].
+    /// let output = Rotated(-2, array);
+    /// ```
+    function Rotated<'T> (stepCount : Int, array : 'T[]) : 'T[] {
```

Comment:
Based on the PR comments, I would've expected this to be replaced by `CircularlyShifted` but currently the change includes both even though they have the same functionality and implementation. Don't we just want to keep one?

Reply:
I am sorry something went wrong between the force push and the merge 😄 should be fine now, please squash at merge